pkg/codegen/hcl2/binder_schema.go

```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
-type packageSchema struct {
+type PackageSchema struct {
```

Comment:
Now that this is exported, do we need to add `// internal` or some such markers so users don't depend on it with the expectation of full support (no breaking changes, etc)? Or this is implied already?

Reply:
> Now that this is exported, do we need to add // internal or some such markers so users don't depend on it with the expectation of full support (no breaking changes, etc)? Or this is implied already?

The canonical way to make this internal would be to factor it out into a new package, `hcl2/internal/whatever`, and export it from that package. I have a _slight_ preference for that approach here, but we kind of already treat these packages as internal interfaces that we're free to break at will, so I don't feel strongly.

Reply:
(and when I say canonical, I mean that the `go` tool would refuse to allow packages outside of `hcl2` to link to `hcl2/internal/whatever`)

Reply:
yikes, yeah I dunno if I want to go down that refactor rabbit hole right now. The issue is really that the program driver needs `BindProgram`, so it imports hcl2. https://github.com/pulumi/pulumi/pull/7657/files#diff-723d6e036de4c1d99d5d402b6dc8988d390ba50413f3d465a7b97056378e42bfR164

And hcl2 needs test.NewHost, so it imports test.

I added a commit that refactors `NewHost` into a new `utils` package to avoid the cycle. Is that preferable to what I had?